pkg/networkservice/chains/nsmgrproxy/server.go

```diff
@@ -22,30 +22,78 @@ import (
 
 	"google.golang.org/grpc"
 
+	"github.com/google/uuid"
 	"github.com/networkservicemesh/api/pkg/api/networkservice"
 
 	"github.com/networkservicemesh/sdk/pkg/networkservice/chains/client"
 	"github.com/networkservicemesh/sdk/pkg/networkservice/chains/endpoint"
 	"github.com/networkservicemesh/sdk/pkg/networkservice/common/connect"
 	"github.com/networkservicemesh/sdk/pkg/networkservice/common/externalips"
 	"github.com/networkservicemesh/sdk/pkg/networkservice/common/interdomainurl"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/common/null"
 	"github.com/networkservicemesh/sdk/pkg/networkservice/common/swapip"
 	"github.com/networkservicemesh/sdk/pkg/tools/token"
 )
 
+type serverOptions struct {
+	name            string
+	authorizeServer networkservice.NetworkServiceServer
+	tokenGenerator  token.GeneratorFunc
+	grpcOptions     []grpc.DialOption
+}
+
+// Option modifies option value
+type Option func(o *serverOptions)
+
+// WithName sets name for the server
+func WithName(name string) Option {
+	return func(o *serverOptions) {
+		o.name = name
+	}
+}
+
+// WithAuthorizeServer sets authorize server for the server
+func WithAuthorizeServer(s networkservice.NetworkServiceServer) Option {
+	return func(o *serverOptions) {
+		o.authorizeServer = s
+	}
+}
+
+// WithTokenGenerator sets token generator function for the server
+func WithTokenGenerator(function token.GeneratorFunc) Option {
+	return func(o *serverOptions) {
+		o.tokenGenerator = function
+	}
+}
+
+// WithGRPCOptions sets gRPC Dial Options for the server
+func WithGRPCOptions(options ...grpc.DialOption) Option {
```

Comment:
Please follow the steps from the issue description.

```suggestion
func WithDialOption(options ...grpc.DialOption) Option {
```

Reply:
Done